Adds Kaltura driver.

scherztc opened the pull request:
Time to merge our Kaltura code and Google code into one master branch.

hortongn commented on app/views/browse_everything/_file.html.erb:
```diff
@@ -37,14 +37,27 @@
+  <td role="gridcell" class="ev-directory-select">
+    <%= check_box_tag(:select_all, "0", false, class: "ev-select-all") if file.container? %>
   </td>
+  <% if provide_name = 'kaltura' %>
```
@scherztc You're not testing equality here.  Should be `==`

hortongn commented on app/views/browse_everything/_file.html.erb:
```diff
@@ -37,14 +37,27 @@
+    <td role="gridcell" class="ev-file-kind">
+      <%= file.type %>
+    </td>  
+  <% end %>
```
Lines 41-60 can probably be refactored to 

```
  <td role="gridcell" class="ev-file-size">
    <%= number_to_human_size(file.size).sub(/Bytes/,'bytes') unless provide_name == 'kaltura' %>
  </td>

  <td role="gridcell" class="ev-file-kind">
    <%= file.type unless provide_name == 'kaltura' %>
  </td>
```

hortongn commented on app/controllers/browse_everything_controller.rb:
```diff
@@ -9,6 +9,11 @@ class BrowseEverythingController < ActionController::Base
 
   def index
+
+    if #{provider_name} == 'kaltura'
```
@scherztc What's going on here?  Why is it commented?

hortongn commented on lib/browse_everything/driver/kaltura.rb:
```diff
@@ -0,0 +1,57 @@
+      end
+
+      def details(path)
+        byebug
```
@scherztc Remove the byebug

hortongn commented on app/controllers/browse_everything_controller.rb:
```diff
@@ -9,6 +9,11 @@ class BrowseEverythingController < ActionController::Base
   def index
+
+    if #{provider_name} == 'kaltura'
+      $current_user = warden.user.email.split("@")[0]
```
@scherztc Can we at least check that the email address is a `uc.edu` address?

scherztc commented:
@hortongn I cleaned up:

Provider equality checks in app/views/browse_everything/_file.html.erb
Conditional refactor in app/views/browse_everything/_file.html.erb
Improved UC username lookup in app/controllers/browse_everything_controller.rb
Removed byebug in app/controllers/browse_everything_controller.rb
